napari/utils/perf/_event.py

```diff
@@ -2,7 +2,14 @@
 """
 import os
 import threading
-from typing import Optional
+from collections import namedtuple
+from typing import Dict, Optional
+
+# A span of time for an event.
```

Comment:
I really like this use of namedtuple, blog your post here https://tobeva.com/articles/brain-compatible-code/ really resonated with me!!